Test Failure (20210815-1105): concurrent.persistent.fat.errorpaths testFailingTaskAndAutoPurgeFEWithPolling

Test Failure (20210815-1105): com.ibm.ws.concurrent.persistent.fat.errorpaths.PersistentExecutorErrorPathsTestWithFailoverAndPollingEnabled.testFailingTaskAndAutoPurgeFEWithPolling

```
junit.framework.AssertionFailedError: 2021-08-15-19:06:55:384 Missing success message in output. PersistentErrorTestServlet is starting testFailingTaskAndAutoPurge<br>
<pre>ERROR in testFailingTaskAndAutoPurge:
javax.persistence.PersistenceException: 
Exception Description: Error binding to externally managed transaction
Internal Exception: javax.transaction.RollbackException: Transaction rolled back
	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:145)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
	at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:104)
	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.create(DatabaseTaskStore.java:267)
	at com.ibm.ws.concurrent.persistent.internal.PersistentExecutorImpl.newTask(PersistentExecutorImpl.java:1384)
	at com.ibm.ws.concurrent.persistent.internal.PersistentExecutorImpl.schedule(PersistentExecutorImpl.java:1622)
	at web.PersistentErrorTestServlet.testFailingTaskAndAutoPurge(PersistentErrorTestServlet.java:134)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at web.PersistentErrorTestServlet.doGet(PersistentErrorTestServlet.java:100)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1258)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:746)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:443)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1227)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5049)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1007)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1159)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:428)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:387)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:566)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:500)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:360)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:327)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:238)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.transaction.RollbackException: Transaction rolled back
	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2287)
	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2268)
	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
</pre>

	at com.ibm.ws.concurrent.persistent.fat.errorpaths.PersistentExecutorErrorPathsTestWithFailoverAndPollingEnabled.runInServlet(PersistentExecutorErrorPathsTestWithFailoverAndPollingEnabled.java:88)
	at com.ibm.ws.concurrent.persistent.fat.errorpaths.PersistentExecutorErrorPathsTestWithFailoverAndPollingEnabled.testFailingTaskAndAutoPurgeFEWithPolling(PersistentExecutorErrorPathsTestWithFailoverAndPollingEnabled.java:180)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:197)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

There appears to be a large slowdown after obtaining the persistence service unit such that by the time it gets to requesting a connection, the transaction has timed out and rolled back, preventing the connection from enlisting.

```
[8/15/21, 19:03:57:416 PDT] 000000b9 id=55691683 ibm.ws.concurrent.persistent.internal.PersistentExecutorImpl > run[poll] Entry 
[8/15/21, 19:03:57:806 PDT] 000000b9 id=014f588d com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore        > findUnclaimedTasks Entry  
[8/15/21, 19:03:57:806 PDT] 000000b9 id=014f588d com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore        > getPersistenceServiceUnitReadUncommitted Entry 
[8/15/21, 19:04:03:744 PDT] 000000b9 id=014f588d com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore        < getPersistenceServiceUnitReadUncommitted Exit  

This is probably not hung here. It likely is only now attmpting the next enlist and gets rejected at that point.
[8/15/21, 19:06:47:634 PDT] 000000b9 id=00000000 com.ibm.ejs.j2c.XATransactionWrapper                         E J2CA0026E: Method addSync caught javax.transaction.RollbackException: Transaction rolled back
	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2287)
	at com.ibm.tx.jta.embeddable.impl.EmbeddableTranManagerSet.registerSynchronization(EmbeddableTranManagerSet.java:229)
	at com.ibm.tx.jta.embeddable.impl.EmbeddableTranManagerSet.registerSynchronization(EmbeddableTranManagerSet.java:216)
	at com.ibm.ws.transaction.services.TransactionManagerService.registerSynchronization(TransactionManagerService.java:406)
	at com.ibm.ejs.j2c.XATransactionWrapper.addSync(XATransactionWrapper.java:170)
	at com.ibm.ejs.j2c.ConnectionManager.initializeForUOW(ConnectionManager.java:973)
	at com.ibm.ejs.j2c.ConnectionManager.involveMCInTran(ConnectionManager.java:692)
	at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:308)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:140)
	at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:114)
	at org.eclipse.persistence.sessions.JNDIConnector.connect(JNDIConnector.java:135)
        ...
	at org.eclipse.persistence.internal.jpa.QueryImpl.getResultList(QueryImpl.java:473)
	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.findUnclaimedTasks(DatabaseTaskStore.java:796)
```

The above is taking almost 3 minutes -- and it hasn't even done anything in the database yet.  Liberty's built-in transaction timeout is 2 minutes.  The test will need to override and greatly increase the built-in transaction timeout if it's going to need to pass under such conditions.

--- a/dev/com.ibm.ws.concurrent.persistent_fat_errorpaths/publish/servers/com.ibm.ws.concurrent.persistent.fat.errorpaths/server.xml
+++ b/dev/com.ibm.ws.concurrent.persistent_fat_errorpaths/publish/servers/com.ibm.ws.concurrent.persistent.fat.errorpaths/server.xml
@@ -1,5 +1,5 @@
 <!--
-    Copyright (c) 2014, 2020 IBM Corporation and others.
+    Copyright (c) 2014, 2021 IBM Corporation and others.
     All rights reserved. This program and the accompanying materials
     are made available under the terms of the Eclipse Public License v1.0
     which accompanies this distribution, and is available at
@@ -42,6 +42,8 @@
   
   <javaPermission codebase="${shared.resource.dir}/derby/derby.jar" className="java.security.AllPermission"/>
   <javaPermission className="java.lang.RuntimePermission" name="getClassLoader"/>
+
+  <transaction totalTranLifetimeTimeout="6m"/>
 
   <variable name="onError" value="FAIL"/>
 </server>

